clang/test/Driver/fsyclbin.cpp

```diff
@@ -0,0 +1,43 @@
+/// Tests behaviors of -fsyclbin
+
+/// -fsyclbin is only used with the new offloading model.
+// RUN: %clangxx -fsycl -fsyclbin --no-offload-new-driver %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefix=UNUSED
+// UNUSED: warning: argument unused during compilation: '-fsyclbin'
+
+/// -fsyclbin -fsycl-device-only usage.  -fsycl-device-only will 'win' and
+/// -fsyclbin is effectively ignored.
+// RUN: %clangxx -fsycl-device-only -fsyclbin --offload-new-driver %s -### 2>&1 \
```

Comment:
What will be the behavior when `-fsycl -fsycl-device-only -fsyclbin` is passed?

Reply:
In the case of `-fsycl -fsycl-device-only -fsyclbin`, `-fsycl-device-only` will win.  Both `-fsycl-device-only` and `-fsyclbin` will imply `-fsycl` so the behavior will be equivalent to using `-fsycl-device-only -fsyclbin`.